build-check-comments.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# Reliable cross-platform comment checker for go files.
+# Uses docker and stable linux distro for reliable grep regex and
+# to make sure we work on Linux and OS X and oh even Windows! :)
+# Main issue is minor incompatibies between grep on various platforms.
+#
+# Linux: set USE_DOCKER=yes to force run in docker
+# All other platforms: uses docker by default. Set USE_DOCKER=no to override.
+#
+# Checks Go code comments if they have trailing period (excludes protobuffers and vendor files).
+# Comments with more than 3 spaces at beginning are omitted from the check, example: '//    - foo'.
+# This does not include top-level commments for funcs and types.
+#
+# Example:
+#   func main() {
+#	    // comment without period, will trigger check
+#	    //comment without leading space, will trigger check.
+#	    // comment without trailing space, will trigger check.
+#	    // good comment, leading space, ends with period, no trailing space.
+#     //     - more than 3 leading spaces, will pass
+#	    app := kingpin.New(filepath.Base(os.Args[0]), "A block storage based long-term storage for Prometheus")
+#   }
+
+
+# Dir of this script
+declare ROOT_DIR=$(cd $(dirname ${BASH_SOURCE}) && pwd)
+declare THIS_SCRIPT="${ROOT_DIR}/$(basename "${BASH_SOURCE}")"
+
+# Image to use if we do docker-based commands, centos is most stable right?
+declare IMAGE="centos:7"
```

Comment:
Maybe I'd use some smaller image if possible? for example some `slim` version of debian?

Reply:
Good suggestion, I was wondering which image to use, busybox didn't work for obvious reasons. Is `debian:9-slim` ok?

Reply:
Yes, that sounds good :+1:

Reply:
Done